### ISSUE TITLE
Update `application/vc+ld+jwt`

### ISSUE THREAD

**Issue by OR13** [w3c/vc-jose-cose#71]
Describe how the media type is different from `application/vc+jwt`.

**Comment by selfissued** [comment]
The media type name is `application/vc+jwt`.  When used as a `typ` or `cty` value, its representation is abbreviated to `vc+jwt`.

**Comment by OR13** [comment]
Proposed text for `application/vc+ld+jwt`

The claimset is expected be a JSON-LD payload conformant to the `vc+ld+json` media type, registered in __reference__.

Proposed text for `application/vc+jwt`

The claimset is expected be a JSON payload following the guidance in https://datatracker.ietf.org/doc/html/rfc8725

**Comment by TallTed** [comment]
> [@selfissued] The media type name is `application/vc+jwt`. When used as a `typ` or `cty` value, its representation is abbreviated to `vc+jwt`.

The second sentence is true, but it is not relevant here, as this issue is not about 
`application/vc+jwt` vs `vc+jwt` 
but about
`application/vc+jwt` vs <code>application/vc<b><i>+ld</i></b>+jwt</code>.

**Comment by TallTed** [comment]
> [@OR13] Proposed text for `application/vc+ld+jwt` / `application/vc+jwt`

Where are the above "proposed texts" proposed to go?

The first "proposed text" should refer to `application/vc+ld+json` (which I believe we still expect to register), not to the abbreviated `vc+ld+json` (which I do not expect to be registered anywhere).

Further, I think both "proposed texts" would be more accurate if they said, <code>The <b>payload</b> is expected to be a `JSON-LD`/`JSON` <b>claimset</b> conformant with ....</code>, or (I think more likely), <code>The <b>payload</b> is expected to be a <b>Verifiable Credential claimset expressed as</b> `JSON-LD`/`JSON` and conformant with ....</code>

**Comment by OR13** [comment]
@TallTed 

Indeed we are intending to register an "application" subtype, with a +jwt structured suffix: 
- https://www.rfc-editor.org/rfc/rfc8417.html#section-7.2

I think I mentioned this before, https://www.rfc-editor.org/rfc/rfc7515#section-4.1.9

```
   To keep messages compact in common situations, it is RECOMMENDED that
   producers omit an "application/" prefix of a media type value in a
   "typ" Header Parameter when no other '/' appears in the media type
   value.  A recipient using the media type value MUST treat it as if
   "application/" were prepended to any "typ" value not containing a
   '/'.  For instance, a "typ" value of "example" SHOULD be used to
   represent the "application/example" media type, whereas the media
   type "application/example;part="1/2"" cannot be shortened to
   "example;part="1/2"".
```

It is also common in the "application type" registry... to omit the `application/` when refering to the sub type.

<img width="1222" alt="Screen Shot 2023-04-06 at 4 41 24 PM" src="https://user-images.githubusercontent.com/8295856/230499276-a7c2d943-b9af-4062-a3d9-155abae6b38c.png">
<img width="1129" alt="Screen Shot 2023-04-06 at 4 41 14 PM" src="https://user-images.githubusercontent.com/8295856/230499288-a167f3e1-aff2-4911-9849-1f9757ddb133.png">

In JWT, the "claimset" is the name given to the "payload" that is secured along with the protected header.

So it is common to see "claimset" and "payload" used interchangeably... The correct term is "claimset" if we are talking about a token format, or a specific type of token, see https://www.rfc-editor.org/rfc/rfc7519#section-4.

```
   The "cty" (content type) Header Parameter is used by JWS applications
   to declare the media type [[IANA.MediaTypes](https://www.rfc-editor.org/rfc/rfc7515#ref-IANA.MediaTypes)] of the secured content
   (the payload).  This is intended for use by the application when more
   than one kind of object could be present in the JWS Payload; the
   application can use this value to disambiguate among the different
   kinds of objects that might be present.
```

- https://www.rfc-editor.org/rfc/rfc7515#section-4.1.10

For `application/vc+ld+jwt`, we would expect `cty` to be `application/vc+ld+json` or `vc+ld+json`.

Per the guidance on explicit typing in https://datatracker.ietf.org/doc/html/rfc8725#name-use-explicit-typing

```
Sometimes, one kind of JWT can be confused for another. 
If a particular kind of JWT is subject to such confusion, 
that JWT can include an explicit JWT type value, 
and the validation rules can specify checking the type. 
This mechanism can prevent such confusion. 
Explicit JWT typing is accomplished by using the "typ" Header Parameter.
```

`application/vc+ld+jwt` is a `typ` value to be paired with the `cty` value of `application/vc+ld+json`.

The intention is to make it clear just by looking at the `typ` header that this is a token that contains JSON-LD, or JSON.

As you noted, we will also want to make it clear that:

`application/vc+jwt` is a `typ` value to be paired with a normal JSON claimset, as described here: https://www.rfc-editor.org/rfc/rfc7519#section-4

The work that needs to be done, is to communicate what is unique or different about these specific `typ` values when compared to a regular `typ: JWT`... Here is some more proposed language to pick at.

Taking your proposed text apart and splitting them by `typ` values, and using the term per the JWT RFC:

#### typ: application/vc+ld+jwt

A `cty` of `application/vc+ld+json` is assumed, and can be omitted. 

The ***claimset*** MUST be a Verifiable Credential expressed as JSON-LD and conformant with [[VC-DATA-MODEL]].

#### typ: application/vc+jwt

A `cty` of `application/json` is assumed, and can be omitted.

The ***claimset*** MUST be a Verifiable Credential expressed as JSON and conformant with [[RFC7519]]. One or more mappings __reference to a mapping__ MAY be applied for processors wishing to consume `application/vc+ld+json`, instead of `application/json`.

**Comment by TallTed** [comment]
@OR13 — Thank you. I think the latest "language to pick at" is an improvement. A little picking —

> A `cty` of `application/vc+ld+json` is assumed, and can be omitted.

and

> A `cty` of `application/json` is assumed, and can be omitted.

If I'm reading the whole of your adjustments correctly, I think it would be better if these say **required** (or **mandatory** or **MUST be** or similar) instead of **assumed**.

I think I could live with these blocks, with the above wording tweak.

**Comment by OR13** [comment]
Thanks!

Usually `cty` is not present when typ JWT is set because it would be `application/json`.

At the face to face it was mentioned that the reason IETF didn't set a `cty` was that they didn't see a need at the time, for specific tokens types or distinguishing different content types for the same token type.

In our case `typ` is sufficient to distinguish between the 2 kinds of token.

If we keep `cty` as a MUST (the way it is in the spec now), it can be ignored safely by processors... I suspect a lot of them will complain when they discover this.

The JWT BCP focuses on typ instead of cty, other folks have pointed this out when we first started discussing media types, and at IETF 116.

This is why I feel it ought to be either a MAY or SHOULD, not a MUST, if normative guidance on `cty` remains.

I would also be comfortable not saying anything about `cty` here, since we now have a good way to relate `vc+ld+json` to `vc+ld+jwt`, via `typ` instead of `cty` and traditional JWTs don't leverage `cty` the way you suggest, as far as I have seen, none of the specific token types, that rely on +jwt structured suffix have mandatory `cty` values, but I could have missed a counter example.

**Comment by OR13** [comment]
I might have misread your comment.

Do you mean:

MUST be omitted.

Or 

MUST be `application/json` / `application/vc+ld+json` (I assumed this above).

**Comment by TallTed** [comment]
Focusing for the moment on "A `cty` of `application/vc+ld+json` is assumed, and can be omitted."

An *assumption* is not the same as "this must always be true."

If the `cty` — the `content type` — MUST be `application/vc+ld+json` when `typ` is `application/vc+ld+jwt`, then that is what should be said.

The `cty` of a resource with `typ` = `application/vc+ld+jwt` is not *assumed* to be `application/vc+ld+json`; rather, if `typ` = `application/vc+ld+jwt` then *by definition*, `cty` MUST be `application/vc+ld+json`, so `cty` need (`MAY`? `SHOULD`? `MUST`?) not be stated when `typ` = `application/vc+ld+jwt`.

**Comment by OR13** [comment]
I like the second framing, especially the last bit.

I will raise a PR when the current one merges to avoid messy conflict.

**Comment by OR13** [comment]
@TallTed this seems like it won't be needed in #88 is merged (as I expect it will be eventually).

I suggest we close this issue, any objection?

**Comment by TallTed** [comment]
This may not be needed when and if #88 is eventually merged. However, that is not a foregone conclusion, i.e., #88 may *not* be eventually merged, and this may still be needed, so this should not be closed until #88 is merged, and it can be confirmed that what is merged as #88 does resolve this issue.

**Comment by Sakurann** [comment]
PR #88 was merged
